native/common/jp_context.cpp

```diff
@@ -0,0 +1,369 @@
+/*****************************************************************************
+   Copyright 2019 Karl Einar Nelson
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+ *****************************************************************************/
+#include "jpype.h"
+#include "jp_typemanager.h"
+#include "jp_arrayclass.h"
+#include "jp_boxedtype.h"
+#include "jp_stringtype.h"
+#include "jp_classloader.h"
+#include "jp_voidtype.h"
+#include "jp_booleantype.h"
+#include "jp_bytetype.h"
+#include "jp_chartype.h"
+#include "jp_shorttype.h"
+#include "jp_inttype.h"
+#include "jp_longtype.h"
+#include "jp_floattype.h"
+#include "jp_doubletype.h"
+#include "jp_reference_queue.h"
+#include "jp_proxy.h"
+
+JPResource::~JPResource()
+{
+}
+
+/*****************************************************************************/
+// Platform handles the differences in dealing with shared libraries
+// on windows and unix variants.
+
+#ifdef WIN32
+#include "jp_platform_win32.h"
+#define  PLATFORM_ADAPTER Win32PlatformAdapter
+#else
+#include "jp_platform_linux.h"
+#define  PLATFORM_ADAPTER LinuxPlatformAdapter
+#endif
+namespace
+{
+
+JPPlatformAdapter* GetAdapter()
+{
+	static JPPlatformAdapter* adapter = new PLATFORM_ADAPTER();
+	return adapter;
+}
+}
+
+
+#define USE_JNI_VERSION JNI_VERSION_1_4
```

Comment:
Quick question: why do you need to pin the version here (I guess it is already set somewhere else?)

Reply:
Not sure.  I think copied this frome somewhere.  Is there a benefit to using a later veesion?

Reply:
I do not know. My suspicion is, that it only matters during JNI_CreateJavaVM, whereas the version is specified. 
I think the definition in jp_javaframe is just not doing anything, so it can be deleted.

Reply:
Okay so it should be set in jp_context and no where else.  This define is used only to get the ENV version when creating the JVM and when attaching threads all of which must happen in context.  The value 1.4 is correct as we do not use any 1.6 features.   We will have to update this with a new value once the JVM adds Value types.

http://cr.openjdk.java.net/~jrose/values/values-0.html


```
jp_context.cpp:#define USE_JNI_VERSION JNI_VERSION_1_4
jp_context.cpp: jniArgs.version = USE_JNI_VERSION;
jp_context.cpp: jint res = m_JavaVM->functions->GetEnv(m_JavaVM, (void**) &env, USE_JNI_VERSION);
jp_context.cpp: return JNI_OK == m_JavaVM->functions->GetEnv(m_JavaVM, (void**) &env, USE_JNI_VERSION);
jp_context.cpp: jint res = m_JavaVM->functions->GetEnv(m_JavaVM, (void**) &env, USE_JNI_VERSION);
```